app/views/layouts/base.html.erb

```diff
@@ -49,11 +49,13 @@
     <![endif]-->
     <% if User.current %>
       <% cache(TopbarSweeper.fragment_name, :skip_digest => true) do %>
-        <div id="layout"></div>
+        <div id="react-app-root"></div>
+        <%= mount_react_app %>
```

Comment:
I wasn't sure so i kept it. can it be deleted?

Reply:
I think you misunderstood my comment - we should cache this data, otherwise we need to recalculate permissions etc every time a page reloads (which would be less with this but still). Another option is to move this caching to the serverside. keep in mind that we might need to change the TopbarSweeper implementation to properly handle the new layout.

Reply:
changing the TopbarSweeper implementation is out of scope, and out of my knowledge right now. Maybe we should create an issue for this.
afaik The Menu items are [cached in the client](https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/components/Layout/LayoutSelectors.js#L14) but i guess you are referring to something else with the permissions. either way I think we should handle this in another PR